.github/workflows/auto_branching.yml

```diff
@@ -0,0 +1,214 @@
+
+            # Perform replacements using a for loop
+            for replacement in "${replacements[@]}"; do
+              old_url=$(echo "$replacement" | cut -d' ' -f1)
+              new_url=$(echo "$replacement" | cut -d' ' -f2)
+              sed "s|${old_url}|${new_url}|g" "$FILE_PATH" > "$TEMP_FILE" && mv "$TEMP_FILE" "$FILE_PATH"
+            done
+
+      - name: Remove the dispatch release GHA
+        id: remove-dispatch-release-gha
+        run: |
+            rm -rf ./.github/workflows/dispatch_release.yml
+            rm -rf ./.github/workflows/auto_branching.yml
+
+      - name: Remove lines with @pytest.mark.stream
+        id: remove-mark-stream
+        run: |
+           # Loop through files in the folder
+           grep -rl "tests/foreman" -e '@pytest\.mark\.stream' | while IFS= read -r file; do
+            awk '!/@pytest\.mark\.stream/' "$file" > temp && mv temp "$file"
+           done
+
+      - name: Update version in setup.py
+        run: sed -i "s/version=['\"][0-9.]*['\"]\+/version='${{ github.event.inputs.target_branch }}'/" setup.py
+
+      - name: Update the Constants in __init__.py file
+        run: |
+          old_url="https://raw.githubusercontent.com/SatelliteQE/robottelo/master/tests/foreman/data/uri.sh"
+          new_url="https://raw.githubusercontent.com/SatelliteQE/robottelo/${{ github.event.inputs.target_branch }}/tests/foreman/data/uri.sh"
+          FILE_PATH="./robottelo/constants/__init__.py"
+
+          awk '/SAT_NON_GA_VERSIONS =/ { sub(/\[[^,]*, /, "[", $0) } 1' "$FILE_PATH" > temp && mv temp "$FILE_PATH"
+          sed -i.bak "s|${old_url}|${new_url}|" "$FILE_PATH"
+          rm "$FILE_PATH.bak"
+
+      - name: git status
+        run: git status
+
+      - name: git diff
+        run: git diff
+
+      - name: Commit changes
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add setup.py ./tests/foreman ./robottelo/* ./requirements.txt ./.github/*
+          git commit -m "Changes for ${{ github.event.inputs.target_branch }} new branch"
+          git remote -vvv
+          git push origin ${{steps.checkout-to-auto-branch.outputs.branch_name}}
+
+      - name: Create pull request
+        run: |
+          title="[${{ github.event.inputs.target_branch }}]: Changes for ${{ github.event.inputs.target_branch }} new branch"
+          body="
+            ### Problem Statement
+            New ${{ github.event.inputs.target_branch }} branch
+
+            ### Solution
+            - Dependabot labels are updated for new branch
+            - Removed dispatch release GHA from ${{ github.event.inputs.target_branch }} as we are releasing only master changes
+            - Airgun and Nailgun Requirements uses ${{ github.event.inputs.target_branch }} branch
+            - Constants are using new version now
+            - Stream tests removed
+            - Setup.py uses new version
+          "
```

Comment:
do you think adding some default PRT test trigger would be good here?

Reply:
That's great addition, I'm thinking to run sanity trigger would be good